### ISSUE TITLE
0 pictures downloaded and command in private chat not working

### ISSUE THREAD

**Issue by IFN3T** [Seklfreak/discord-image-downloader-go#88]

**Comment by smokeofc** [comment]
Also having this problem.

ran it first to gen the config file, edited the file with user credentials, and put up a few channels under the [channels] section.

When launching it changes the playing state of the user to '0 pictures downloaded' but doesn't actually seem to download anything, nor does it respond to the commands 'help' or 'history'.

**Comment by Rouskuh** [comment]
I can confirm, private chat does not work

**Comment by Kakkela** [comment]
Could you show your interactive channels config

**Comment by smokeofc** [comment]
[auth]
email    = m**@gmail.com
password = **

[general]
skip edits                           = true
download tistory sites               = false
max download retries                 = 5
download timeout                     = 60
send notices to interactive channels = false

[channels]
324727917026213890 = G:\Discord\Test1
324727896340037653 = G:\Discord\Test2
358552198017318922 = G:\Discord\Test3

[flickr]
api key = your flickr api key

[twitter]
consumer key        = your consumer key
consumer secret     = your consumer secret
access token        = your access token
access token secret = your access token secret

**Comment by Kakkela** [comment]
Okay so with that kind of setup it won't respond to commands, only way to send commands to it is Interactive channels. It should download from the channels in your _[channels]_ section though. If the account has permission to view the channels.

For help and history you need to follow these instructions:

> By default, the tool only downloads new links posted while the tool is running. You can also set up the tool to download the complete history of a channel. To do this you have to run this tool with a separate discord account. Send your second account a dm on your primary account and get the channel id from the direct message channel. Now add this channel id to the config by adding the following lines:
> 
```
[interactive channels]
<your channel id> = <some valid path>
```
> After this is done restart the tool and send history as a DM to your second account. The bot will ask for the channel id of the channel you want to download and start the downloads. You can view all available commands by sending help.
>
> Where do I get the channel id?
Open discord in your browser and go to the channel you want to monitor. In your address bar should be a URL like https://discordapp.com/channels/1234/5678. The number after the last slash is the channel id, in this case, 5678. Or, enable Developer mode and right click the channel you need, and click Copy ID.

Note that you need to do the last part by logging into Discord with your downloader account and then open DM with your main account.

**Comment by smokeofc** [comment]
I have two accounts, one running the selfbot, one normal one.
Sending the commands 'help' and 'history' to the selfbot account doesn't give a response, and no downloading is happening as new posts are posted to the channels either.

**Comment by Kakkela** [comment]
Your configuration is not set to take commands via DM, you need the **[interactive channels]** section for it to work.

For not getting images from regular channels, make sure you have appropriate permissions to view those channels with your selfbot account. If bot does have access to that channel and you post image there it should show up on console window at least.

**Comment by Kakkela** [comment]
```
[auth]
email    = 
password =  

[general]
skip edits                           = false
download tistory sites               = true
max download retries                 = 5
download timeout                     = 60
send notices to interactive channels = true

[interactive channels]
<id> = <path>

[channels]
<id> = <path>
<id> = <path>
<id> = <path>
```

So this ignoring the twitter and flickr parts and specific options i have set on **[general]**, this is how your config should look like.
The ID in interactive channels part should be the direct message ID from your selfbot account TO your main account. There's guide for obtaining that in my earlier reply. If you try to get this ID with right clicking user in DM list it isnt the correct ID. It need's to come from selfbot account side TO your main account side with the method descibed above.

**Comment by smokeofc** [comment]
![2018-06-08 01_40_32-config ini - notepad](https://user-images.githubusercontent.com/13959015/41131680-13a1249c-6abd-11e8-87a6-f9687ea76d20.png)
This should match what you said, but it still doesn't do anything in DM....
(Didn't know what ID to use, as either side of the chat produces a different ID, so used each one by itself, and both as in the picture, with no result)

**Comment by Makein** [comment]
Same thing here...

**Comment by Kakkela** [comment]
And you are absolute sure those IDs are obtained with the URL method? If so there is no reason why that shouldn't work.

**Comment by Makein** [comment]
Yeah, browser url

**Comment by Makein** [comment]
Can you help thought discord view?

**Comment by Kakkela** [comment]
We can take a look at it later tomorrow maybe.

**Comment by smokeofc** [comment]
well... fml...
I got the ID by using dev mode (right-click, copy id)... getting it from the browser fixed my problem...
sry bout the trouble... everything's ok now...

**Comment by Makein** [comment]
And I still not getting it.
+ eng not my main language and I can misunderstood something...

**Comment by IFN3T** [comment]
after defined [interactive channels] it's working.

thank.

**Comment by Makein** [comment]
HOW?

**Comment by smokeofc** [comment]
OK, Makein found me on discord, so helped him out there. for future readers, here are some instructions:

1. When you've downloaded the files, put the discord-image-downloader-go-windows-amd64.exe file in a folder, let's say 'C:\DiscordDL' for the purposes of this instruction, but you can choose your own location.
2. You will see a command window open and close as it's generating a skeleton config file that you can use to config the selfbot. it will look like so:
`[auth]
email    = your@email.com
password = your password

[general]
skip edits                           = true
download tistory sites               = false
max download retries                 = 5
download timeout                     = 60
send notices to interactive channels = false

[channels]
channelid1 = C:\full\path\1
channelid2 = C:\full\path\2
channelid3 = C:\full\path\3

[flickr]
api key = your flickr api key

[twitter]
consumer key        = your consumer key
consumer secret     = your consumer secret
access token        = your access token
access token secret = your access token secret`
change 'send notices to interactive channels = false' to 'send notices to interactive channels = true'
3. fill out your email and password.
4. Create a new section somewhere in the file with the heading [interactive channels], you need that to be able to 'chat' with your secondary account.
5. in a browser, open https://discordapp.com/, and login with your secondary account.
6. You'll be in the DM view, open a chat with your main account, and you'll notice that the URL is something along the lines of https://discordapp.com/channels/@me/873685642264840742.
7. copy the last number from the url and paste it in the config file under [interactive channels], and add a location like so:
`[interactive channels]
873685642264840742 = C:\DiscordDL\DM`
8. now you're going to need the id of the channel you want to download from, the easiest way to do so is to enable developer mode. You can do so in either the main client or the browser view by going to the discord settings, under Appearance. After activating it, right click on the channel you want and select Copy ID.
9. Paste it under [channels], and include a download location, like so:
`[channels]
232143587603644416 = C:\DiscordDL\Server\Channel`

If you've done as instructed, when you write 'history' (without ') in a DM to your secondary account, you should get the following response:
`Please tell me one or multiple Channel IDs (separated by commas)
Please make sure the channels have been whitelisted before submitting.`
Just paste a ID that's already listed under [channels] into the chat, and it should start downloading every image posted to that channel

If you leave the selfbot running, it will automatically download all new images added to all channels listed under [channels], and if you only want to launch it every now and then to update, it seems to ignore pictures that you've already downloaded, so just run the 'history' command again.

Hope this helps anyone still having trouble setup'ing this thing :)
I think this issue can be closed now.

**Comment by YumekuiChaser** [comment]
Hi, I have the same issue, I have done all mentioned before and still it does nothing :(

**Comment by smokeofc** [comment]
You'll prolly have to use login token instead of email and password these days, but the rest should work still, you sure you've followed my guide to the letter?

**Comment by EvocativeOlash** [comment]
smokeofc's explanation is much more clear than the one on the main page.  Perhaps there's a language barrier, but I could only understand what to do after reading their explanation.

thanks!

**Comment by YumekuiChaser** [comment]
No, I use mail and password and when I launch the command line says its logged as <<my username>> but when I use the history command it does nothing in the DM

**Comment by schokapyk** [comment]
Thanks @smokeofc it worked 👍
